spring-cloud-vault-core/src/main/java/org/springframework/cloud/vault/VaultClient.java

```diff
@@ -102,6 +124,94 @@ public VaultClientResponse write(URI uri, Object entity, VaultToken vaultToken)
 				createHeaders(vaultToken)));
 	}
 
+	/**
+	 * Initialize the vault using with {@code createKeys} and {@code requiredKeys}
+	 *
+	 * @param createKeys The number of keys available to unseal
+	 * @param requiredKeys The number of keys required to unseal
+	 * @return A {@link VaultInitializedResponse} containing the initialized state of vault.
+	 */
+	public VaultInitializedResponse initialize(int createKeys, int requiredKeys) {
+		VaultInitializeRequest initializeVault = VaultInitializeRequest.of(createKeys, requiredKeys);
+
+		ResponseEntity<VaultInitializedResponse> initResponse = restTemplate.exchange(
+				buildUri(INITIALIZE_URL_TEMPLATE), HttpMethod.PUT,
+				new HttpEntity<>(initializeVault), VaultInitializedResponse.class);
+
+		if (!initResponse.getStatusCode().is2xxSuccessful()) {
+			throw new IllegalStateException("Cannot initialize vault: "
+					+ initResponse.toString());
+		}
+		return initResponse.getBody();
+
+	}
+
+
+	/**
+	 * Unseal the vault using the given {@code key}
+	 *
+	 * @param key must not be {@literal null}.
+	 * @return A {@link VaultSealStatusResponse} containing the current seal status.
+	 */
+	public VaultSealStatusResponse unseal(String key) {
+		Assert.notNull(key, "key must not be empty!");
+		Map<String, String> requestBody = new HashMap<String, String>();
+		requestBody.put("key", key);
+		ResponseEntity<VaultSealStatusResponse> unsealResponse = restTemplate.exchange(
+				buildUri(UNSEAL_URL_TEMPLATE), HttpMethod.PUT, new HttpEntity<>(requestBody),
+				VaultSealStatusResponse.class);
+		return unsealResponse.getBody();
+	}
+
+
+	/**
+	 * Query the vault for it's current seal status
+	 *
+	 * @return A {@link VaultSealStatusResponse} containing the current seal status.
+	 */
+	public VaultSealStatusResponse sealStatus() {
+		ResponseEntity<VaultSealStatusResponse> unsealResponse = restTemplate.exchange(
+				buildUri(SEAL_STATUS_URL_TEMPLATE), HttpMethod.GET, new HttpEntity<>(null, createHeaders()),
+				VaultSealStatusResponse.class);
+		return unsealResponse.getBody();
+	}
+
+	/**
+	 * Query the current Vault service for it's health status
+	 *
+	 * @return A {@link VaultHealthResponse} containing the current service status.
+	 */
+	public VaultHealthResponse health() {
```

Comment:
I'm not sure I agree.  As a user of the extension I would like to interact with Vault through a single client (my assumption is that this would be VaultClient).  In the absence of using Actuator or even outside of the HealthIndicator I may still find need to check the health of the Vault service.  I think this plays into a discussion at the end of the PR.